src/h5web/visualizer/Visualizer.tsx

```diff
@@ -1,20 +1,22 @@
-import React, { ReactElement, useState } from 'react';
+import React, { ReactElement, useState, useContext } from 'react';
 import { AsyncResourceContent } from 'use-async-resource';
 import type { HDF5Entity } from '../providers/models';
 import styles from './Visualizer.module.css';
-import { useSupportedVis } from './hooks';
+import { getSupportedVis } from './utils';
 import { VIS_DEFS, Vis } from '../visualizations';
 import VisSelector from './VisSelector';
 import Loader from './Loader';
+import { ProviderContext } from '../providers/context';
 
 interface Props {
   entity?: HDF5Entity;
 }
 
 function Visualizer(props: Props): ReactElement {
   const { entity } = props;
+  const { metadata } = useContext(ProviderContext);
 
-  const supportedVis = useSupportedVis(entity);
+  const supportedVis = entity ? getSupportedVis(entity, metadata) : [];
```

Comment:
We talked about it I think, I'm turning the hook into a simple function and I move a bit of the responsibility out of it and into `Visualizer`. This makes it more testable and just easier to read I think.